senza/components/taupage_auto_scaling_group.py

```diff
@@ -1,38 +1,62 @@
+import json
+import re
+import sys
+import textwrap
 
 import click
 import pierone.api
-import textwrap
 import yaml
-import json
-import sys
-import re
-
 from senza.aws import resolve_referenced_resource
 from senza.components.auto_scaling_group import component_auto_scaling_group
 from senza.docker import docker_image_exists
 from senza.utils import ensure_keys
-
+from zign.api import get_existing_token
 
 _AWS_FN_RE = re.compile(r"('[{]{2} (.*?) [}]{2}')", re.DOTALL)
 
 
 def check_docker_image_exists(docker_image: pierone.api.DockerImage):
+    token = None
     if 'pierone' in docker_image.registry:
-        try:
-            exists = pierone.api.image_exists('pierone', docker_image)
-        except pierone.api.Unauthorized:
+        token = get_existing_token('pierone')
```

Comment:
OK, might be, so it would just work for human users --- you can use zign.api.get_token(...) which tries best effort and also works with service tokens